### PR TITLE
DT-929 adding permissions/role check for create, update and deletion of court hearings.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
As part of discussions with the NART team it was decided the permissions to create. retrieve and update court hearings needs to be hardened.

### CODE REVIEW

File: src/main/java/net/syscon/elite/service/CourtHearingCancellationService.java [hunk 1]
```diff
@@ -23,6 +24,7 @@ public CourtHearingCancellationService(final CourtEventRepository courtEventRepo
     @Transactional
     @VerifyBookingAccess
     @HasWriteScope
+    @PreAuthorize("hasRole('PECS_PRISON')")
```
Comment (andymarke) [inline src/main/java/net/syscon/elite/service/CourtHearingCancellationService.java:27]: Very minor should this be something like 'COURT_HEARING_MAINTAINER' which is assigned to the PECS system credentials

Comment (matt-crowson-moj) [reply src/main/java/net/syscon/elite/service/CourtHearingCancellationService.java:27]: Hmmm not sure as this what came about talking with @Mjwillis

Comment (matt-crowson-moj) [reply src/main/java/net/syscon/elite/service/CourtHearingCancellationService.java]: Updated.

File: src/test/java/net/syscon/elite/api/resource/impl/OffenderMovementsResourceImplIntTest_cancelCourtHearing.java [hunk 2]
```diff
@@ -42,7 +42,7 @@ public Clock clock() {
 
     @Before
     public void setup() {
-        token = authTokenHelper.getToken(AuthTokenHelper.AuthToken.NORMAL_USER);
+        token = authTokenHelper.getToken(AuthTokenHelper.AuthToken.COURT_HEARING_MAINTAINER);
```
Comment (petergphillips) [inline src/test/java/net/syscon/elite/api/resource/impl/OffenderMovementsResourceImplIntTest_cancelCourtHearing.java:45]: would be good to test the negation too i.e. I could delete all of the `PreAuthorize` lines from the code and all the tests would still pass

Comment (matt-crowson-moj) [reply src/test/java/net/syscon/elite/api/resource/impl/OffenderMovementsResourceImplIntTest_cancelCourtHearing.java:45]: Good point.  Will add.

**Pull request merged**